sdk/core/Azure.Core.Experimental/tests/Variant/StoringBoolean.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using NUnit.Framework;
+
+namespace Azure
+{
+    public class StoringBoolean
+    {
+        [TestCase(true)]
+        [TestCase(false)]
+        public void BooleanImplicit(bool testValue)
+        {
+            Value value;
+            using (MemoryWatch.Create)
+            {
+                value = testValue;
+            }
+
+            Assert.AreEqual(testValue, value.As<bool>());
+            Assert.AreEqual(typeof(bool), value.Type);
+
+            bool? source = testValue;
+            using (MemoryWatch.Create)
+            {
+                value = source;
+            }
+            Assert.AreEqual(source, value.As<bool?>());
+            Assert.AreEqual(typeof(bool), value.Type);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void BooleanCreate(bool testValue)
+        {
+            Value value;
+            using (MemoryWatch.Create)
+            {
+                value = Value.Create(testValue);
+            }
+
+            Assert.AreEqual(testValue, value.As<bool>());
+            Assert.AreEqual(typeof(bool), value.Type);
+
+            bool? source = testValue;
+
+            using (MemoryWatch.Create)
+            {
+                value = Value.Create(source);
+            }
+
+            Assert.AreEqual(source, value.As<bool?>());
+            Assert.AreEqual(typeof(bool), value.Type);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void BooleanInOut(bool testValue)
+        {
+            Value value;
+            bool success;
+            bool result;
+
+            using (MemoryWatch.Create)
+            {
+                value = new(testValue);
+                success = value.TryGetValue(out result);
+            }
+
+            Assert.True(success);
+            Assert.AreEqual(testValue, result);
+
+            Assert.AreEqual(testValue, value.As<bool>());
+            Assert.AreEqual(testValue, (bool)value);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void NullableBooleanInBooleanOut(bool? testValue)
+        {
+            bool? source = testValue;
+            Value value;
+            bool success;
+            bool result;
+
+            using (MemoryWatch.Create)
+            {
+                value = new(source);
+                success = value.TryGetValue(out result);
+            }
+
+            Assert.True(success);
+            Assert.AreEqual(testValue, result);
+
+            Assert.AreEqual(testValue, value.As<bool>());
+
+            Assert.AreEqual(testValue, (bool)value);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void BooleanInNullableBooleanOut(bool testValue)
+        {
+            bool source = testValue;
+            Value value = new(source);
+            bool success = value.TryGetValue(out bool? result);
```

Comment:
Do we expect this to allocate ... is that why it's not in a MemoryWatch using block?

Reply:
This should not allocate. The reason this is not watched is that it would make the test more complex (because of using statement scoping rules) but the goal of the test is to test the result, not to test allocations.